### PR TITLE
Added Linear Search [C]

### LINKED ISSUE #85: Linear Search [C]
https://en.wikipedia.org/wiki/Linear_search

### PR DESCRIPTION
**Pull request by PiyushPawar17**
fixes #85 

#### By submitting this pull request I confirm I've read and complied with the below requirements.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

### CODE REVIEW

File: linear_search/linear_search.c [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+*/
+
+int linear_search(int *arr, int size, int search_element) {
+
+	int i;
+	for (i = 0; i < size; i++) {
+		if (arr[i] == search_element)
+			return i;				// Element found
+	}
+
```
Comment (aashutoshrathi) [inline linear_search/linear_search.c]: remove unwanted blank line

File: linear_search/linear_search.c [hunk 2]
```diff
@@ -0,0 +1,40 @@
+#include <stdio.h>
+
+/*
+*  Algorithm to find a given number in an array
+*  *arr : given array to search in
+*  search_element : Element to be searched
+*  size : size of array
+*/
+
+int linear_search(int *arr, int size, int search_element) {
```
Comment (singhpratyush) [inline linear_search/linear_search.c]: Can be `const int *arr`.

File: linear_search/linear_search.c [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	for (i = 0; i < size; i++) {
+		if (arr[i] == search_element)
+			return i;				// Element found
+	}
+	return -1;						// Element not found
+}
+
+int main() {
+
+	int array[100],i;
```
Comment (singhpratyush) [inline linear_search/linear_search.c]: Place a space after comma.

File: linear_search/linear_search.c [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+int main() {
+	int array[100], i;
+
+	// Initializing array with 1,2,....,100
+	for (i = 1; i <= 100; i++)
+		array[i-1] = i;
+
+	// Elements to be searched in the array
+	int search[5] = {12,55,34,102,78};
```
Comment (jsroyal) [inline linear_search/linear_search.c]: Give space after "," in the array.

File: linear_search/linear_search.c [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+	int i;
+	for (i = 0; i < size; i++) {
+		if (arr[i] == search_element)
+			return i;				// Element found
+	}
+	return -1;						// Element not found
+}
+
+int main() {
+	int array[100], i;
```
Comment (jsroyal) [inline linear_search/linear_search.c]: Inconsistent space.Only one tab or four space.

File: linear_search/linear_search.c [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+	// Initializing array with 1,2,....,100
+	for (i = 1; i <= 100; i++)
+		array[i-1] = i;
+
+	// Elements to be searched in the array
+	int search[5] = {12,55,34,102,78};
+
+	for (i = 0; i < 5; i++) {
+		int index = linear_search(array, 100, search[i]);
```
Comment (jsroyal) [inline linear_search/linear_search.c]: All the variables declaration just after main function.

File: linear_search/linear_search.c (hunk 5 above)
Comment (singhpratyush) [inline linear_search/linear_search.c]: Tabs are consistent.

File: linear_search/linear_search.c (hunk 6 above)
Comment (singhpratyush) [inline linear_search/linear_search.c]: Here is the best scope for `index`, hence change isn't needed.

Comment (jsroyal) [inline linear_search/linear_search.c]: But in the style of c code it should be.

File: linear_search/linear_search.c (hunk 5 above)
Comment (jsroyal) [inline linear_search/linear_search.c]: but there are two tabs. Is that correct way?

Comment (singhpratyush) [inline linear_search/linear_search.c]: > there are two tabs

I see only one.

File: linear_search/linear_search.c (hunk 6 above)
Comment (singhpratyush) [inline linear_search/linear_search.c]: In the style of programming, scope for variables should be kept as small as possible.

File: linear_search/linear_search.c (hunk 4 above)
Comment (singhpratyush) [inline linear_search/linear_search.c]: @PiyushPawar17: Waiting for this change.

**Comment by aviaryan** [comment]
@jsroyal Please update your review.

**Comment by singhpratyush** [comment]
@jsroyal @aashutoshrathi: Please update your review.

**Comment by aviaryan** [comment]
Merged as @aashutoshrathi 's comment had been labelled "outdated" and was fixed by the OP.

**Pull request merged**